File: src/ros2_bridge/mappers/sensor_msgs/channel_float32.rs
//! Typed mapper for `sensor_msgs/msg/ChannelFloat32`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn channel_float32_to_bus(msg: ros_env::sensor_msgs::msg::ChannelFloat32) -> crate::sensor_msgs::msg::v1::ChannelFloat32 {
    crate::sensor_msgs::msg::v1::ChannelFloat32 {
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
        values: crate::ros2_bridge::mappers::convert::f32_seq(msg.values),
    }
}

pub(crate) fn channel_float32_to_ros(bus: crate::sensor_msgs::msg::v1::ChannelFloat32) -> ros_env::sensor_msgs::msg::ChannelFloat32 {
    ros_env::sensor_msgs::msg::ChannelFloat32 {
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
        values: bus.values,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct SensorMsgsChannelFloat32Mapper;

impl TypedTopicMapper for SensorMsgsChannelFloat32Mapper {
    type Ros = ros_env::sensor_msgs::msg::ChannelFloat32;
    type Bus = crate::sensor_msgs::msg::v1::ChannelFloat32;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(channel_float32_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(channel_float32_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/sensor_msgs/channel_float32.py
"""Generated mapper for `sensor_msgs/msg/ChannelFloat32`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.sensor_msgs.msg.v1 import ChannelFloat32 as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def channel_float32_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.name = str(msg.name)
    bus.values.extend(list(msg.values))
    return bus


def channel_float32_to_ros(bus):
    from sensor_msgs.msg import ChannelFloat32 as RosMsg

    out = RosMsg()
    out.name = str(bus.name)
    out.values = list(bus.values)
    return out


class SensorMsgsChannelFloat32Mapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from sensor_msgs.msg import ChannelFloat32 as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return channel_float32_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return channel_float32_to_ros(bus)
